web/controllers/folder_controller.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Ask.FolderController do
+  use Ask.Web, :api_controller
+
+  alias Ask.{Folder, Logger, ActivityLog}
+  alias Ecto.Multi
+
+  def create(conn, params = %{"project_id" => project_id}) do
+    folder_params = Map.get(params, "folder", %{})
+                      |> Map.put("project_id", project_id)
+
+    %Folder{}
+    |> Folder.changeset(folder_params)
+    |> Repo.insert()
+    |> case do
+      {:ok, folder} ->
+        conn
+        |> put_status(:created)
+        |> render("show.json", folder: folder)
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        conn
+        |> put_status(:unprocessable_entity)
+        |> render(Ask.ChangesetView, "error.json", changeset: changeset)
+    end
+  end
+
+  def index(conn, %{"project_id" => project_id}) do
+    folders = (from f in Folder,
+          where: f.project_id == ^project_id)
+    |> Repo.all
+
+    conn
+    |> render("index.json", folders: folders)
+  end
+
+  def show(conn, %{"id" => folder_id}) do
+    folder = Folder
+    |> Repo.get!(folder_id)
+
+    conn
+    |> render("show.json", folder: folder)
+  end
+
+  def delete(conn, %{"project_id" => project_id, "id" => folder_id}) do
+    conn
+    |> load_project_for_change(project_id)
+
+    folder = (from f in Folder,
+      where: f.id == ^folder_id)
+    |> Repo.one
+    |> Repo.preload(:surveys)
```

Comment:
Why this `preload`?

Reply:
It's obsolete. Loading the folder surveys in order to check if it's empty before deletion was beaten by this approach:
https://github.com/instedd/surveda/pull/1466/files#diff-0b5c9ff98396655c4ce233e07312219bR25